### DISCUSSION TITLE
What allows streamlink to pause live streams in VLC?

### DISCUSSION THREAD

**Discussion by Serenmew** [streamlink/streamlink#4921] [Q&A] [answered]
Hey all, hoping to get a bit of light shed on this since I feel like I've hit a brick wall. Opening a stream through the command prompt using streamlink allows you to pause and unpause whenever you'd like. This is extremely useful for me in my applications. However simply opening the same stream via the open network stream option in VLC doesnt allow pausing of any sort. How does streamlink accomplish this?

The reason I ask to give more background, is that I want to sync timers on two separate twitch streams from within obs using the VLC Media Source. This by default performs just like opening a network stream. I'm wondering if it's just settings that streamlink is setting in VLC every time it opens an instance of it, because I remember reading once that VLC Media Source uses your VLC settings to an extent. Also just to add a tiny bit more context, I have tried using the same links within the Media Source built into OBS, however this doesnt seem to properly buffer videos and is relatively useless for my applications.

Thank you :)

**Comment by bastimeyer** [comment, marked as answer]
Unless you set specific options, Streamlink will simply set the player's input to `stdin` (standard input stream), nothing else. Both Streamlink and your player have separate buffers. If your player's playback is paused and it doesn't continue reading from Streamlink's buffer, then Streamlink's own buffer gets filled until it either gets full or until the player starts reading data from it again. See the `--ringbuffer-size` docs:
https://streamlink.github.io/cli.html#cmdoption-ringbuffer-size

I don't use VLC, so I can't tell you how to change its own internal network streaming logic when pausing or why it does certain things the way it does.

There's also a third-party project for using Streamlink as OBS input directly, so you don't have to capture any rendered window contents and audio channels, but I haven't used it yet myself:
https://streamlink.github.io/thirdparty.html#obs-streamlink

**Reply by Serenmew** [reply]
I see, thank you! I'm not sure how I can apply this to fix my problem yet but at least now I understand a bit better. Hopefully I can find some way to implement this buffer or figure out how to get VLC's working. In theory I've read it should be able to pause live media but I cant quite figure it out yet.

I have tried that third party project but I definitely wouldnt recommend it. I could only get it working in OBS 26.1, it uses an outdated version of streamlink and doesnt have any media controls. It also broke my install of 26.1 and I had to reinstall OBS on top of itself to even get it to launch.
